pkg/cloudprovider/vsphere/nodemanager_test.go

```diff
@@ -0,0 +1,227 @@
+	var InsecureFlag bool
+	var err error
+	cfg.Global.VCenterIP = os.Getenv("VSPHERE_VCENTER")
+	cfg.Global.VCenterPort = os.Getenv("VSPHERE_VCENTER_PORT")
+	cfg.Global.User = os.Getenv("VSPHERE_USER")
+	cfg.Global.Password = os.Getenv("VSPHERE_PASSWORD")
+	cfg.Global.Datacenters = os.Getenv("VSPHERE_DATACENTER")
+	cfg.Network.PublicNetwork = os.Getenv("VSPHERE_PUBLIC_NETWORK")
+	if os.Getenv("VSPHERE_INSECURE") != "" {
+		InsecureFlag, err = strconv.ParseBool(os.Getenv("VSPHERE_INSECURE"))
+	} else {
+		InsecureFlag = false
+	}
+	if err != nil {
+		log.Fatal(err)
+	}
+	cfg.Global.InsecureFlag = InsecureFlag
+
+	ok = (cfg.Global.VCenterIP != "" &&
+		cfg.Global.User != "")
+
+	return
+}
+
+// configFromSim starts a vcsim instance and returns config for use against the vcsim instance.
+// The vcsim instance is configured with an empty tls.Config.
+func configFromSim() (Config, func()) {
+	return configFromSimWithTLS(new(tls.Config), true)
+}
+
+// configFromSimWithTLS starts a vcsim instance and returns config for use against the vcsim instance.
+// The vcsim instance is configured with a tls.Config. The returned client
+// config can be configured to allow/decline insecure connections.
+func configFromSimWithTLS(tlsConfig *tls.Config, insecureAllowed bool) (Config, func()) {
+	var cfg Config
+	model := simulator.VPX()
+
+	err := model.Create()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	model.Service.TLS = tlsConfig
+	s := model.Service.NewServer()
+
+	// STS simulator
+	path, handler := sts.New(s.URL, vpx.Setting)
+	model.Service.ServeMux.Handle(path, handler)
+
+	// Lookup Service simulator
+	model.Service.RegisterSDK(lookup.New())
+
+	cfg.Global.InsecureFlag = insecureAllowed
+
+	cfg.Global.VCenterIP = s.URL.Hostname()
+	cfg.Global.VCenterPort = s.URL.Port()
+	cfg.Global.User = s.URL.User.Username()
+	cfg.Global.Password, _ = s.URL.User.Password()
+	cfg.Global.Datacenters = vclib.TestDefaultDatacenter
+	cfg.Network.PublicNetwork = vclib.TestDefaultNetwork
+	cfg.VirtualCenter = make(map[string]*VirtualCenterConfig)
+	cfg.VirtualCenter[s.URL.Hostname()] = &VirtualCenterConfig{}
+
+	return cfg, func() {
+		s.Close()
+		model.Remove()
+	}
+}
+
+// configFromEnvOrSim returns config from configFromEnv if set, otherwise returns configFromSim.
+func configFromEnvOrSim() (Config, func()) {
+	cfg, ok := configFromEnv()
+	if ok {
+		return cfg, func() {}
+	}
+	return configFromSim()
+}
+
+func TestRegUnregNode(t *testing.T) {
+	cfg, ok := configFromEnvOrSim()
+	defer ok()
+
+	vsphere, err := newVSphere(cfg)
+	if err != nil {
+		t.Fatalf("Failed to construct/authenticate vSphere: %s", err)
+	}
+
+	nm := NodeManager{
+		vsphereInstanceMap: vsphere.vsphereInstanceMap,
+		nodeNameMap:        make(map[string]*NodeInfo),
+		nodeUUIDMap:        make(map[string]*NodeInfo),
+		nodeRegUUIDMap:     make(map[string]*v1.Node),
+	}
+
+	name := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine).Config.GuestFullName
```

Comment:
Since you're grabbing multiple fields, should use the same vm for both:
```go
vm := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
name := vm.Config.GuestFullName
id := vm.Config.Uuid
```

Reply:
Ahhhh I didn't realize that would grab multiple VM instances per call. Will update.